utils.py

```diff
@@ -62,6 +74,56 @@ def load_image_for_dataset(filename):
     }
 
 
+def load_dataset(folder, img_ext='.png'):
+    data = []
+    data_mask = []
+    for datum in dataset_from_folder(folder, img_ext):
+        data.append(datum['img'])
+        data_mask.append(datum['mask'])
+    return np.asarray(data), np.asarray(data_mask)
+
+
+def calc_PCA(data):
+    # normalize data before calculating PCA
+    if (data > 1).any():
+        norm_data = data.astype('float32') / 255.0
+    else:
+        norm_data = data
+    n, h, w, c = norm_data.shape
+    reshaped_data = np.reshape(norm_data, (n * h * w, 3))
+    conv = reshaped_data.T.dot(reshaped_data) / reshaped_data.shape[0]
+    u, s, v = np.linalg.svd(conv)
+    eigenvalues = np.sqrt(s)
+    return eigenvalues, u
+
+
+def blur_batch(batch):
+    blured_batch = []
+    for image in batch:
+        gray_im = cv.cvtColor(image, cv.COLOR_BGR2GRAY)
+        blur_amount = cv.Laplacian(gray_im, cv.CV_64F).var()/1000.
```

Comment:
@Foxiy I nearly missed the `1000.` on this line, I believe that for the sake of clarity it should really say `1000.0`.